apiserver/facades/client/resources/repository.go

```diff
@@ -112,14 +223,22 @@ func resourceFromRevision(rev transport.ResourceRevision) (charmresource.Resourc
 	if err != nil {
 		return charmresource.Resource{}, errors.Trace(err)
 	}
+	fingerprint, err := charmresource.ParseFingerprint(rev.Download.HashSHA384)
+	if err != nil {
+		return charmresource.Resource{}, errors.Trace(err)
+	}
 	r := charmresource.Resource{
 		Meta: charmresource.Meta{
 			Name: rev.Name,
 			Type: resType,
+			// TODO (stickupkid): API is missing both of these fields.
```

Comment:
transport ResourceRevision now has the data, landed with 12422, however the data will not be filled in yet.